### PR TITLE
Add fuzzing

### PR DESCRIPTION
**Pull request by stevenjohnstone**
Fuzzing for yp_parse_serialize, yp_regexp_named_capture_group_names and yp_unescape_manipulate_string. Automation of common tasks via make and docker plus some helper scripts.

### CODE REVIEW

File: fuzz/parse.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+OUTPUT_DIR=$1
+
+screen -S grammar_fuzzer -d -m  /bin/bash -c "AFL_CUSTOM_MUTATOR_LIBRARY=/usr/local/lib/libgrammarmutator-ruby.so AFL_CUSTOM_MUTATOR_ONLY=1 afl-fuzz -i fuzz/corpus/parse -o $OUTPUT_DIR -S grammar -- ./build/fuzz.parse"
```
Comment (kddnewton) [inline fuzz/parse.sh:5]: Seems like the parse corpus is a direct copy of the tests from the top level. Could we do something to remove this duplication like mount that as a directory in docker instead?

Comment (stevenjohnstone) [reply fuzz/parse.sh:5]: Yes, that'd work. Or a symlink. I was wondering if there's a better collection of programs I can use as a corpus? This was the laziest possible thing I could do...

Comment (kddnewton) [reply fuzz/parse.sh:5]: We should probably use our fixtures. They're the best corpus we have. Everything under `test/fixtures/**/*.txt` is a Ruby program. We have them listed as text only to not confuse tools looking for all Ruby programs.

Comment (stevenjohnstone) [reply fuzz/parse.sh:5]: That's far more sensible! I'll try that out.

File: fuzz/dict [hunk 2]
```diff
@@ -0,0 +1,31 @@
+"__ENCODING__"
```
Comment (kddnewton) [inline fuzz/dict:1]: Could we expand this dictionary to include at least one of every token? For the most part you can just copy the examples out of `config.yml`

**Comment by kddnewton** [comment]
This looks really good @stevenjohnstone, thanks so much! I'm going to merge this and I'd like to see if we can iterate on their grammar.json file because it looks relatively bare-bones.

**Pull request merged**